nomos-services/data-availability/network/src/backends/mod.rs

```diff
@@ -25,11 +28,19 @@ pub trait NetworkBackend<RuntimeServiceId> {
         overwatch_handle: OverwatchHandle<RuntimeServiceId>,
         membership: Self::Membership,
         addressbook: Self::Addressbook,
+        subnets_settings: SubnetsConfig,
     ) -> Self;
     fn shutdown(&mut self);
     async fn process(&self, msg: Self::Message);
     async fn subscribe(
         &mut self,
         event: Self::EventKind,
     ) -> Pin<Box<dyn Stream<Item = Self::NetworkEvent> + Send>>;
+    async fn start_historic_sampling(
+        &self,
+        block_number: BlockNumber,
+        blob_id: BlobId,
+        membership: Vec<(PeerId, Multiaddr)>,
+    );
+    fn local_peer_id(&self) -> PeerId;
```

Comment:
This is not the API this should have. Ideally we should pass here the set of blob_ids for a block and the membership. Nothing else. This is not ready yet in the underlaying behaviours/swarm. So not sure if we should stall this meanwhile.